src/core/common/instance.hpp

```diff
@@ -745,6 +750,13 @@ template <> inline DuaManager &Instance::Get(void)
 }
 #endif
 
+#if (OPENTHREAD_MTD || OPENTHREAD_FTD) && OPENTHREAD_CONFIG_MLR_ENABLE
```

Comment:
nit: remove `(OPENTHREAD_MTD || OPENTHREAD_FTD)`

This is within `#if OPENTHREAD_MTD || OPENTHREAD_FTD` block already.

https://github.com/openthread/openthread/blob/3a0569e96542c8a428ee13ba4f7961099f316d42/src/core/common/instance.hpp#L410-L769